src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.ProjectFormat/SdkStyleConversionSolutionWideStep.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.ProjectFormat
+{
+    public class SdkStyleConversionSolutionWideStep : UpgradeStep
+    {
+        private readonly TryConvertRunner _runner;
+
+        private IEnumerable<UpgradeStep>? _substeps;
+
+        public override string Description => $"Use the try-convert tool ({_runner.Path}{_runner.VersionString}) to convert the projects in a solution";
+
+        public override string Title => $"Convert projects in a solution to SDK style";
+
+        public override string Id => WellKnownStepIds.TryConvertProjectConverterStepId;
```

Comment:
This is interesting. My first reaction was that this should have a unique ID, but I think sharing is actually better in this case since this step is 'subbing in' for the old one (maybe replacing it eventually?) and other steps that depended on the old one can just depend on this transparently without needing the complication of checking feature flags, etc.

Reply:
exactly why I did it this way. It's conceptually the same thing and is mutually exclusive with the other.